### PR TITLE
Added few more guidelines for good pattern for swagger.

### PR DESCRIPTION
**Pull request by anudeepsharma**

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@anudeepsharma__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (ans). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: documentation/swagger-good-patterns.md [hunk 1]
```diff
@@ -13,10 +13,18 @@
  - Checks on the Resource Model
     - Every "Resource" Model should have "location" as a "required" property as per the mandate from ARM team; no exceptions are allowed over here.
     - Every "Resource" Model should be tagged with "x-ms-azure-resource". This will cause the Resource to inherit from the Resource definition in the client runtime.
+    - GET/PUT/PATCH for a resource should return the same "Resource" Model, also PUT should accept the same body (there can be readonly, optional, required properties).
  - Consistency in the naming of list methods
```
Comment (amarzavery) [inline documentation/swagger-good-patterns.md]: It is worthwhile mentioning over here that the body parameter for a PATCH operation should have all optional properties and they should not have default values.

Comment (anudeepsharma) [inline documentation/swagger-good-patterns.md]: For PATCH I have doubts what we should do, structure wise the Resource Body is good, but all values should be nullable, so for int it does not matter that whether in swagger there is any default value or not, or whether that is  optional or not. I think for PATCH code generation should take care of generating the same structure as of Resource body but with nullable types. For Example if a Resource is like { poolQuota: int, size: string }, structure generated for PUT request/response, GET response, PATCH response should be { int PoolQuota, string Size} where as for PATCH request it should be { int? PoolQuota, string Size }. In swagger for PATCH request same body should be there as for PUT.

I am not sure for this pull request we should make any updates on PATCH.

Comment (anudeepsharma) [inline documentation/swagger-good-patterns.md]: Added good practice for PATCH operation as well.

File: documentation/swagger-good-patterns.md [hunk 2]
```diff
@@ -13,10 +13,19 @@
...
     - Operations that provide resources in a subscription should be named as "OperationGroup_List"
     - If the ListByResourceGroup is overloaded "i.e. if the resourcegroup is null then the operation lists all resources in the subscription" then, such operations should not be allowed.
+ - Consistency of the Names of the APIs, parameters and models
+    - Resource Model name should be same as singular form of the resource type. Like for virtualMachines resource type it should be VirtualMachine.
+    - Parameter names in the apis should be name of the resource like virtualMachines\{virtualMachine} or virtualMachines\{virtualMachineName} both are good example but virtualMachines\{vm} is bad example. 
+    - Any action name should be same as what is defined in the Rest APIs, like regenerateAdminKeys should be regenerateAdminKeys in the Swagger as well, where as regenerateKeys is bad example.
+ - Check for Completeness
+    - Ask for all operations information from the RP, which is usaully can be get from https://management.azure.com/providers/{serviceName}/operations?api-version={apiVersion}.
```
Comment (amarzavery) [inline documentation/swagger-good-patterns.md]: Ask for ~~all operations~~ information about all the operations from a RP, which ~~is usually~~ usually can be obtained by making a GET request where the uri is of the following pattern:
- `https://management.azure.com/providers/{serviceName}/operations?api-version={apiVersion}.`

Comment (anudeepsharma) [inline documentation/swagger-good-patterns.md]: Done.

**Pull request merged**